### PR TITLE
chore: logging fix

### PR DESCRIPTION
**Pull request by whynowy**
Fix logging in main.go

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -110,34 +108,43 @@ func main() {
...
+	if err != nil {
+		logger.Fatalw("Unable to set GitSync controller", zap.Error(err))
+	}
 
+	// Watch GitSync objects
+	if err := gitSyncController.Watch(&source.Kind{Type: &apiv1.GitSync{}}, &handler.EnqueueRequestForObject{},
+		predicate.Or(
+			predicate.GenerationChangedPredicate{}, predicate.LabelChangedPredicate{},
+		)); err != nil {
+		logger.Fatalw("Unable to watch GitSyncs", zap.Error(err))
```
Comment (juliev0) [inline cmd/main.go:135]: Interesting. I'd actually added the GenerationChanged Predicate in [this](https://github.com/numaproj-labs/numaplane/pull/54/files) PR but I guess it was removed at some point.

Comment (juliev0) [reply cmd/main.go:135]: Do you have any type of Label specifically in mind as far as watching for Label changes as well?

Comment (juliev0) [reply cmd/main.go:135]: Actually, regarding my first comment, I noticed the code is still in there: `reconciler.SetupWithManager()` does actually look for the `GenerationChangedPredicate{}` still

Comment (whynowy) [reply cmd/main.go]: Reverted!

File: Dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Build the manager binary
-FROM golang:1.20 as builder
+FROM golang:1.21 as builder
```
Comment (juliev0) [inline Dockerfile:2]: I guess that would've also taken care of [this](https://github.com/numaproj-labs/numaplane/pull/149) issue, right? So, now we should be good to use go 1.21 as long as we all are running that version locally

**Review by juliev0 — APPROVED** [review]

**Pull request merged**